src/Compilers/Helpers/GlobalAssemblyCacheHelpers/GacFileResolver.cs

```diff
@@ -11,37 +11,46 @@
 namespace Microsoft.CodeAnalysis.Scripting.Hosting
 {
     /// <summary>
-    /// Extends MetadataFileReferenceResolver to enable resolution of assembly
-    /// simple names in the GAC.
+    /// Resolves assembly identities in Global Assembly Cache.
     /// </summary>
     internal sealed class GacFileResolver : IEquatable<GacFileResolver>
     {
         /// <summary>
-        /// Architecture filter used when resolving assembly references.
+        /// Returns true if GAC is available on the platform.
         /// </summary>
-        public ImmutableArray<ProcessorArchitecture> Architectures { get; }
+        public static bool IsAvailable => CoreClrShim.AssemblyLoadContext.Type == null;
```

Comment:
In general we should be asking whether a specific feature is available, not on which platform we are. 

It's true that in this case an absence of AssemblyLoadContext doesn't neccessarily mean presence of GAC, so I guess a better way to check would be to try GetProcAddress on the clr.dll entry points that implement factories for GAC COM interfaces. I can follow up on that later. For now check for AssemblyLoadContext is close enough.